### PR TITLE
Fixes #19898: launch set-up wizard on first connection

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/19898

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/SetupRedirect.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import com.normation.rudder.web.model.CurrentUser
+import com.normation.box._
+import com.normation.rudder.AuthorizationType.Administration
+import net.liftweb.http.CurrentReq
+
+class SetupRedirect extends DispatchSnippet with Loggable {
+
+  private[this] val configService = RudderConfig.configService
+  private[this] val eventLogRepo  = RudderConfig.eventLogRepository
+  private[this] val uuidGen  = RudderConfig.stringUuidGenerator
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/SetupRedirect.scala:53]: this two vals (eventLogRepo & uuidGen) cause a warning: never used

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/SetupRedirect.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  def display() = {
+
+    configService.rudder_setup_done().toBox match {
+      case Full(false) if CurrentUser.checkRights(Administration.Write) && !CurrentReq.value.request.url.contains("administration/setup") =>
+         RedirectTo("/secure/administration/setup")
+      case Full(_) =>
+        Noop
+      case eb : EmptyBox =>
+        val msg = eb ?~! "Could not get 'metrics' property"
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/SetupRedirect.scala:67]: should be "setup done" in place of metrics

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala [hunk 3]
```diff
@@ -224,6 +224,8 @@ trait ReadConfigService {
 
   def rudder_generation_continue_on_error(): IOResult[Boolean]
 
+  def rudder_setup_done(): IOResult[Boolean]
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala:227]: I think it should no be a boolean are, but a (at least) three case: 
- not done
- remind me latter
- done

Comment (fanf) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala:227]: (with remind me storing the iso-date at which we should show again the pop-up)

**Review by fanf — COMMENTED** [review]
Appart for the non used vals and the question for type of setup done, everything LGTM.

**Comment by VinceMacBuche** [comment]
Commit modified

**Review by fanf — APPROVED** [review]

**Pull request merged**